core/templates/dev/head/services/ImprovementCardService.js

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for consolidating and accessing the types of
+ * improvement cards that can be rendered in the Improvements Tab.
+ *
+ * All Improvement Cards must provide the following functions (which are
+ * enforced by unit tests):
```

Comment:
The "enforced by unit tests" part is important ... but where are the tests? I would expect to see them in ImprovementCardServiceSpec.js (using a for loop to retrieve all the different card types so that, if a new card type is added, the tests automatically run for that card).